client/pages/projects/[projectId]/editProject.jsx

```diff
@@ -0,0 +1,251 @@
+import React, { useEffect, useState, useCallback } from "react";
+import { useRouter } from "next/router";
+import Link from "next/link";
+import Select from "react-select";
+import classnames from "classnames";
+
+import {
+  Col,
+  Button,
+  Row,
+  Container,
+  Alert,
+  Nav,
+  NavItem,
+  NavLink
+} from "reactstrap";
+import {
+  getModelsByID,
+  canEdit,
+  addModelStage,
+  deleteForm,
+  updateProject
+} from "../../../utils/apiWrapper";
+import {
+  Head,
+  Stage,
+  Loader,
+  AddStage,
+  WrappedMessage
+} from "../../../components";
+import groupSizeData from "../../../utils/groups";
+
+import "../../../public/styles/editProject.scss";
+
+const capitalize = str =>
+  str.length > 0 ? str.charAt(0).toUpperCase() + str.slice(1) : str;
+
+export default WrappedMessage(function EditProjectPage(props) {
+  const [project, setProject] = useState(null);
+  const [isOwner, setIsOwner] = useState(false);
+  const [loading, setLoading] = useState(true);
+  const [activePhase, setActivePhase] = useState("inspiration");
+
+  const [projTitle, setProjTitle] = useState("");
+  const [description, setDescription] = useState("");
+  const [grpSize, setGrpSize] = useState(false);
+
+  const router = useRouter();
+  const { projectId } = router.query;
+
+  const handleTitleChange = projTitle => {
+    setProjTitle(projTitle.target.value);
+  };
+
+  const handleDescriptionChange = description => {
+    setDescription(description.target.value);
+  };
+
+  const handleGrpSizeChange = grpSize => {
+    setGrpSize(grpSize);
+  };
+
+  const deleteProject = id => {
+    deleteForm(id);
+  };
+
+  const saveTopChanges = (name, description, groupSize) => {
+    updateProject(projectId, name, description, groupSize.label)
+      .then(() => props.setSuccess("Successfully updated"))
+      .catch(() => props.setError("Failed to update project"));
+  };
+
+  const loadProject = useCallback(async () => {
+    const project = await getModelsByID(projectId);
+    setProject(project.data);
+    setLoading(false);
```

Comment:
nitpick but can we put any instances of `setLoading(false)` at the end of their corresponding function body so that everything else happens first (lower chance of causing content jumping)